worlds/tunic/__init__.py

```diff
@@ -297,8 +324,48 @@ def remove_filler(amount: int) -> None:
             if tunic_item.name in slot_data_item_names:
                 self.slot_data_items.append(tunic_item)
 
+        # pull out the filler so that we can place it manually during pre_fill
+        self.local_filler = []
+        if self.options.grass_randomizer and self.options.grass_fill > 0 and self.multiworld.players > 1:
+            # skip items marked local or non-local, let fill deal with them in its own way
+            all_filler = []
+            non_filler = []
+            for item in tunic_items:
+                if item.classification in [ItemClassification.filler, ItemClassification.trap] and item.name not in self.options.local_items and item.name not in self.options.non_local_items:
+                    all_filler.append(item)
+                else:
+                    non_filler.append(item)
+            amount_to_local_fill = int(self.options.grass_fill.value * len(all_filler) / 100)
+            for _ in range(amount_to_local_fill):
+                self.local_filler.append(all_filler.pop())
+            tunic_items = all_filler + non_filler
```

Comment:
This could probably be combined into a single loop if you wanted. It's a small optimization but if there are 5700 local items to fill and over 6000 items already being looped through, then just making that a single loop of ~6000 could help 🤷 
```suggestion
            offworld_filler = []
            non_filler = []
    for item in tunic_items:
        if item.classification in [ItemClassification.filler, ItemClassification.trap] and item.name not in self.options.local_items and item.name not in self.options.non_local_items:
            if local_fill_count < amount_to_local_fill:
                self.local_filler.append(item)
                local_fill_count += 1
            else:
                offworld_filler.append(item)
        else:
            non_filler.append(item)
    
    tunic_items = offworld_filler + non_filler
```

Reply:
```suggestion
            all_filler = []
            non_filler = []
            amount_to_local_fill = int(self.options.grass_fill.value * len(all_filler) / 100)
            for item in tunic_items:
                if item.classification in [ItemClassification.filler, ItemClassification.trap] and item.name not in self.options.local_items and item.name not in self.options.non_local_items:
                    if len(self.local_filler) < amount_to_local_fill:
                        self.local_filler.append(item)
                    else:
                        all_filler.append(item)
                else:
                    non_filler.append(item)

            tunic_items = all_filler + non_filler
```
Just a slight change from hesto's version to use len() instead of keeping a separate count (since I'm pretty sure the length of a list is stored in python so it should be slightly faster)
Definitely good suggestion there, hesto!